openapi.go

```diff
@@ -459,6 +459,38 @@ func parseStructTags(t reflect.Type, schemaRef *openapi3.SchemaRef) {
 		propertyCopy := *property
 		propertyValue := *propertyCopy.Value
 
+		if field.Type.Kind() == reflect.Struct {
+			parseStructTags(field.Type, schemaRef.Value.Properties[jsonFieldName])
+		}
+
+		// basic Xml tag support
+		//
+		// We support the following xml tags:
+		// - "xml" field tag: specifies the XML element name
+		// - "xml=attr" field tag: specifies that the field is an XML attribute
+		// - "xml=wrapped" field tag: specifies that the field is wrapped in an XML element
+		xmlField := field.Tag.Get("xml")
+		// if xml tag name is "-", don't add it to the schema
+		if xmlField != "-" && xmlField != "" {
+			xmlFieldName := strings.Split(xmlField, ",")[0] // remove omitempty, etc
+
+			if xmlFieldName == "" {
+				xmlFieldName = field.Name
+			}
+
+			propertyValue.XML = &openapi3.XML{
+				Name: xmlFieldName,
+			}
+
+			xmlFields := strings.Split(xmlField, ",")
+			if slices.Contains(xmlFields, "attr") {
+				propertyValue.XML.Attribute = true
+			}
+			if slices.Contains(xmlFields, "wrapped") {
+				propertyValue.XML.Wrapped = true
+			}
+		}
```

Comment:
I would move this in a dedicated method, and also move the code related to json tag parsing to its own method too.

Because now the code is too complex.

Also it would allow you to return with early return when needed. It would avoid imbricated if levels